packages/flutter/lib/src/widgets/draggable_scrollable_sheet.dart

```diff
@@ -298,12 +353,35 @@ class _DraggableScrollableSheetState extends State<DraggableScrollableSheet> {
     _extent = _DraggableSheetExtent(
       minExtent: widget.minChildSize,
       maxExtent: widget.maxChildSize,
+      snap: widget.snap,
+      snapSizes: _impliedSnapSizes(),
       initialExtent: widget.initialChildSize,
       listener: _setExtent,
     );
     _scrollController = _DraggableScrollableSheetScrollController(extent: _extent);
   }
 
+  List<double> _impliedSnapSizes() {
+    widget.snapSizes?.asMap().forEach((int index, double snapSize) {
```

Comment:
Dart generally says to avoid `forEach` unless you're applying a function to it directly. This should be rewritten as a regular for loop.